Orange/widgets/visualize/owboxplot.py

```diff
@@ -270,14 +277,13 @@ def set_data(self, dataset):
         self.attribute = None
         if dataset:
             domain = dataset.domain
-            self.group_vars[:] = \
-                [None] + \
-                [a for a in chain(domain.variables, domain.metas)
-                 if a.is_discrete]
-            self.attrs[:] = chain(domain.variables,
-                                  (a for a in domain.metas if a.is_primitive()))
+            self.group_vars.set_domain(domain)
+            self.attrs.set_domain(domain)
             if self.attrs:
-                self.attribute = self.attrs[0]
+                # Select the first non-class variable; if all are classes,
+                # select the first class
+                self.attribute = \
+                    self.attrs[len(domain.class_vars) % len(self.attrs)]
```

Comment:
I don't get the formula (`len(domain.class_vars) % len(self.attrs)`)

How does this "select the first non-class variable; if all are classes, select the first class"

Reply:
Let's say that there are 5 classes and 2 other attributes (or metas). Then `len(domain.class_vars)` is 5 and `len(self.attrs)` is 7, so `5 % 7` is 5 and you get the first non-class variable.

If there are 5 classes and 0 other attributes, `len(domain.class_vars)` and `len(self.attrs)` equal 5, so `5 % 5`  = 0 and you get the first class.